src/features/admin/routes/item/edit-item.tsx

```diff
@@ -0,0 +1,46 @@
+import ItemForm from "./item-form";
+import {useStrongParams} from "../../../../hooks/use-strong-params";
+import {BaseLayout} from "../../../../components";
+import {useCreateItemDataQuery} from "../../hooks/use-create-item-data-query";
+import {useUpdateItemMutation} from "../../hooks/use-update-admin-mutation";
+import {queryClient} from "../../../../lib";
+import {ItemData} from "../../types/table-data";
+import { useNavigate } from "react-router-dom";
+
+function EditItem() {
+
+    const navigate = useNavigate();
+    const {itemId} = useStrongParams("itemId");
+    const {data: item} = useCreateItemDataQuery(parseInt(itemId, 10));
+
+    const handleSubmit = (item: ItemData) => {
+        function removeUnusedVariants(item: ItemData) {
```

Comment:
This should ideally not be function and also they should not mutate the input data.

Let's add an TODO so that we refactor this to be immutable (e.g. by using `produce` from [immer](https://immerjs.github.io/immer/)) and also to remove the functions.

Reply:
added todo